### ISSUE TITLE
Sonoff Zigbee3.0 USB Dongle pairing issues

### ISSUE THREAD

**Issue by notenoughtech** [Koenkk/zigbee2mqtt#9117]
I got serious pairing issues with various Zigbee 3.0 end devices. Sonoff Zigbee3.0 USB Dongle was using 
{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20210120,"transportrev":2},"type":"zStack3x0"}
and the latest firmware after a successful flash. 
With the latest firmware, I was not able to get anything to pair - no MQTT messages about the interview process. So I reverted back to the 20210120 revision and I was able to pair Aqara TVOC and IKEA button, but Sonoff devices or Xiaomi buttons were not even entering the interview stage. The same devices were tested with ZZH and CC2531 and I was able to pair and use the devices.

There isn't much information I can share, as the log wasn't being populated. 
Happy to try things out

**Comment by notenoughtech** [comment]
On this firmware I see pairing messages -- so there is progress

Some observations:
AqaraE1:  Paired without even triggering the pairing procedure (was it waiting to pair for 2 days?)
Sonoff button: - no dice not responsive:
`Oct 11 19:16:13 testpi npm[2173]: Zigbee2MQTT:error 2021-10-11 19:16:13: Failed to configure '0x00124b001cd61409', attempt 3 (Error: Bind 0x00124b001cd61409/1 genOnOff from '0x00124b0022a54d11/1' failed (AREQ - ZDO - bindRsp after 10000ms)
`
and also triggered
`
Oct 11 19:11:02 testpi npm[2173]: (node:2188) UnhandledPromiseRejectionWarning: AssertionError [ERR_ASSERTION]: No image available for imageType '14976'
Oct 11 19:11:02 testpi npm[2173]:     at getImageMeta (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/lib/ota/zigbeeOTA.js:24:5)
Oct 11 19:11:02 testpi npm[2173]:     at processTicksAndRejections (internal/process/task_queues.js:95:5)
Oct 11 19:11:02 testpi npm[2173]:     at isNewImageAvailable (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/lib/ota/common.js:265:18)
Oct 11 19:11:02 testpi npm[2173]:     at Object.isUpdateAvailable (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/lib/ota/common.js:256:23)
Oct 11 19:11:02 testpi npm[2173]:     at OTAUpdate.onZigbeeEvent (/opt/zigbee2mqtt/lib/extension/otaUpdate.ts:61:31)
`


Sonoff Motion sensor - at first I thought it would fail but took 2 min to pair and works ok now 🤷‍♂️

Lidl stuff, before not able to pair but worked just like that.

Sonoff temp sensor:  gets stuck on 
Oct 11 19:36:43 testpi npm[2173]: Zigbee2MQTT:info  2021-10-11 19:36:43: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"description":"Temperature and humidity sensor","friendly_name":"0x00124b002269c2c8","model":"SNZB-02","supported":true,"vendor":"SONOFF"},"type":"pairing"}'

gave up after 5 min

Some improvement there over what it was before  - I wasn't on the launchpad version. I need to setup sniffing to give you more sets, got lots on my plate this week so I will try to post as soon as I can

**Comment by notenoughtech** [comment]
Any suggestions for big delays in pairing? I noticed stopping system also
took some time with this stick.

**Comment by notenoughtech** [comment]
This sound very familiar if I'm honest

**Comment by hoggerz** [comment]
Hi Koen,

I’ve recently purchased a couple of these, straight away with the latest 3.X.0 branch firmware, and also with the newer one you posted specifically referencing the sonoff dongle, I found issues with pairing Ikea bulbs. Oddly some that had older firmware versions paired first time with no issues, but newer firmware bulbs always had issues either pairing or setting up reporting for the bulbs. I had similar issues with some very old HUE b22 bulbs  problems I will sniff the traffic of various items when I’m back home at the weekend.

hoggerz

**Comment by u8915055** [comment]
Id like to leave a comment here as well.  I purchased a couple of the sonoff zigbee 3.0 plus dongles and have not been able to get them to work.  Now im new to zigbee2mqtt.. ive been a long time ZHA user through HA but want to move to z2m to get better device support.

I am running this on a debian vm, fresh install.  I followed all the instructions to install z2m.  I took my dongle and flashed it with 20210708.  I was able to get z2m started and the web interface came up etc.  It says its listening for devices to join but i cant get a single device to join.  No logs no nothing (even with debug logging enabled)  It just sits there.  I have flashed this dongle on both a windows vm and also using the CC2538-bsl script.. both seemed to flash and verify successfully but still wont enable any devices to join.  

I have about 25 zigbee devices that are all running fine with ZHA.  I am running on channel 20.

I have found though that ive run into that problem with starting z2m raises that duplicate pan_id issue.. I was able to get around that by re-flashing and setting the pan_id to GENERATE in teh config file.  That sometimes works.  What always seems to let me get things started is to remove the external antenna.  But that fact sort of suggests to me that something is happening at the stack level because with the antenna installed i hit the pan_id issue and without z2m will start.

Anyway i know im not giving a lot of cold hard facts here but that's where things  stand for me.  Unable to get a single device to join.  Im thinking of flashing a CC2531 device and see if i can get that to accept devices on the same setup.

Happy to test or try anything.  Thanks in advance.

**Comment by u8915055** [comment]
Wanted to add some more info here.. i took my other sonoff 3.0 dongle from its package, disconnected the antenna.. replaced the flashed one, and restarted zigbee2mqtt, and then reattached the antenna.  I am now able to get a few devices to connect.. it says in the web interface that its running firmware 20210708 as well.. strange why this one works and the other one doesnt.  One may say i flashed it wrong but i did it a few times and did full verification on the written image and it checked out.  Im going to take the original one and use it with ZHA and see if it works fine there.  

Now im having other issues iwth it reading battery levels, not sure whats happening there, but at least its reading some devices.

**Comment by Hedda** [comment]
> Wanted to add some more info here.. i took my other sonoff 3.0 dongle from its package, disconnected the antenna.. replaced the flashed one, and restarted zigbee2mqtt, and then reattached the antenna. I am now able to get a few devices to connect.. it says in the web interface that its running firmware 20210708 as well.. **strange why this one works and the other one doesnt.** One may say i flashed it wrong but i did it a few times and did full verification on the written image and it checked out. Im going to take the original one and use it with ZHA and see if it works fine there.

Sounds like a hardware fault with the other one of your adapters if one adapter is working and the other adapter is not when both are flashed with same firmware image. Maybe try switching antenna between them to rule out one of the antennas being at fault. 

FYI, once fixed know there is more general and broader discussion about ITead's Sonoff Zigbee 3.0 USB Dongle Plus here -> https://github.com/Koenkk/zigbee2mqtt/discussions/8840

**Comment by Nobody931** [comment]
Hello, I ordered the Sonoff Zigbee3.0 USB Dongle (ITEAD) two weeks ago and it arrived yesterday. Before I used the CC2530 with serial connection the my raspberry Pi4 which worked good, however, after updating the raspberry pi I was not able to connect the CC2530, so I order the Sonoff Dongle (as I read that it does not perform that good anymore).
I was able to connect the Dongle and pair with a xiaomi motion detector. However, I am **not able to pair** my other devices which compromises of IKEA light bulbs, e1743 buttons and other supported devices. I found this issue and before I try reflashing the USB Dongle (which already shipped with the latest firmware [20210708]) I thought maybe there is another problem and used the cc2531 sniffer for capturing the communication. This is the capture of the e1743 button with a fresh battery.
[logAndCapture.zip](https://github.com/Koenkk/zigbee2mqtt/files/7654335/logAndCapture.zip)
I hope the log and capture is good enough, I necessary I can provide more log and capture other devices as well if time allows it :) Would love the get my old setup working, thank you!
tldr: Changed from CC2530 to pre-flashed Sonoff Dongle, successfully paired motion sensor, but not able to pair other (supported and functioning) devices.

**Comment by u8915055** [comment]
﻿
Yes I found the sonoff dongle just doesn’t work as well as my old setup with a sonoff zhabridge with tasmota.  I really think zigbee2mqtt looks really interesting but I can’t get it working with this adapter. I just ordered another adapter, from here. https://slae.sh/projects/cc2652/#get-one   I’m going to try it with this.  Hopefully I get a better outcome.  

For the record I have about 25 devices and with the sonoff 3.0 stick I tried about six of them. Four came in fine and two would not pick up the battery state.  I tried many times to re pair them.  No luck.  All these devices worked fine with ZHa.

**Comment by flatsiedatsie** [comment]
My Sonoff 3.0 Plus USB stick arrived. I've been trying it on a fresh Linux device (Webthings Gateway), with a brand new random PAN_ID and Network_Key.

I tried to switch an IKEA light over from my main system (with Conbee2), to this new test system.

So far I have not been able to get it to pair.

I have tried:
- powering off the main controller with the Conbee2 stick, just to make sure it wasn't interfering.
- moving the Sonoff dongle and lightbulb closer together
- many reboots and full power-off situations.

At some point I noticed an error:

```
2021-12-20 22:19:19.703 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:debug 2021-12-20 22:19:19: Received MQTT message on 'zigbee2mqtt/bridge/request/permit_join' with data '{"value": true}'
2021-12-20 22:19:19.705 INFO   : zigbee2mqtt-adapter: 
2021-12-20 22:19:19.707 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:info  2021-12-20 22:19:19: Zigbee: allowing new devices to join.
2021-12-20 22:19:19.709 INFO   : zigbee2mqtt-adapter: 
2021-12-20 22:19:19.712 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:debug 2021-12-20 22:19:19: Received MQTT message on 'zigbee2mqtt/bridge/config/devices/get' with data ''
2021-12-20 22:19:19.713 INFO   : zigbee2mqtt-adapter: 
2021-12-20 22:19:25.714 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:error 2021-12-20 22:19:25: Request 'zigbee2mqtt/bridge/request/permit_join' failed with error: 'SRSP - ZDO - mgmtPermitJoinReq after 6000ms'
2021-12-20 22:19:25.717 INFO   : zigbee2mqtt-adapter: 
2021-12-20 22:19:25.772 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:debug 2021-12-20 22:19:25: Error: SRSP - ZDO - mgmtPermitJoinReq after 6000ms
2021-12-20 22:19:25.774 INFO   : zigbee2mqtt-adapter:     at Timeout.waiter.timer.setTimeout [as _onTimeout] (/home/pi/.webthings/data/zigbee2mqtt-adapter/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
2021-12-20 22:19:25.776 INFO   : zigbee2mqtt-adapter:     at ontimeout (timers.js:436:11)
2021-12-20 22:19:25.778 INFO   : zigbee2mqtt-adapter:     at tryOnTimeout (timers.js:300:5)
2021-12-20 22:19:25.779 INFO   : zigbee2mqtt-adapter:     at listOnTimeout (timers.js:263:5)
2021-12-20 22:19:25.781 INFO   : zigbee2mqtt-adapter:     at Timer.processTimers (timers.js:223:10)
2021-12-20 22:19:25.782 INFO   : zigbee2mqtt-adapter: 
2021-12-20 22:19:25.784 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:info  2021-12-20 22:19:25: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{},"error":"SRSP - ZDO - mgmtPermitJoinReq after 6000ms","status":"error"}'
```

But most of the time there's... nothing.

I still have to try:
- extension cable
- pairing different devices

**Comment by Koenkk** [comment]
@flatsiedatsie make sure to check https://www.zigbee2mqtt.io/guide/faq/#zigbee2mqtt-crashes-after-some-time

**Comment by dreesi** [comment]
> My Sonoff 3.0 Plus USB stick arrived. I've been trying it on a fresh Linux device (Webthings Gateway), with a brand new random PAN_ID and Network_Key.
> 
> I tried to switch an IKEA light over from my main system (with Conbee2), to this new test system.
> 
> So far I have not been able to get it to pair.
> 
> I have tried:
> 
> * powering off the main controller with the Conbee2 stick, just to make sure it wasn't interfering.
> * moving the Sonoff dongle and lightbulb closer together
> * many reboots and full power-off situations.
> 
> At some point I noticed an `SRSP - ZDO - mgmtPermitJoinReq after 6000ms`error:
> 
> ```
> 2021-12-20 22:19:19.703 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:debug 2021-12-20 22:19:19: Received MQTT message on 'zigbee2mqtt/bridge/request/permit_join' with data '{"value": true}'
> 2021-12-20 22:19:19.705 INFO   : zigbee2mqtt-adapter: 
> 2021-12-20 22:19:19.707 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:info  2021-12-20 22:19:19: Zigbee: allowing new devices to join.
> 2021-12-20 22:19:19.709 INFO   : zigbee2mqtt-adapter: 
> 2021-12-20 22:19:19.712 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:debug 2021-12-20 22:19:19: Received MQTT message on 'zigbee2mqtt/bridge/config/devices/get' with data ''
> 2021-12-20 22:19:19.713 INFO   : zigbee2mqtt-adapter: 
> 2021-12-20 22:19:25.714 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:error 2021-12-20 22:19:25: Request 'zigbee2mqtt/bridge/request/permit_join' failed with error: 'SRSP - ZDO - mgmtPermitJoinReq after 6000ms'
> 2021-12-20 22:19:25.717 INFO   : zigbee2mqtt-adapter: 
> 2021-12-20 22:19:25.772 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:debug 2021-12-20 22:19:25: Error: SRSP - ZDO - mgmtPermitJoinReq after 6000ms
> 2021-12-20 22:19:25.774 INFO   : zigbee2mqtt-adapter:     at Timeout.waiter.timer.setTimeout [as _onTimeout] (/home/pi/.webthings/data/zigbee2mqtt-adapter/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
> 2021-12-20 22:19:25.776 INFO   : zigbee2mqtt-adapter:     at ontimeout (timers.js:436:11)
> 2021-12-20 22:19:25.778 INFO   : zigbee2mqtt-adapter:     at tryOnTimeout (timers.js:300:5)
> 2021-12-20 22:19:25.779 INFO   : zigbee2mqtt-adapter:     at listOnTimeout (timers.js:263:5)
> 2021-12-20 22:19:25.781 INFO   : zigbee2mqtt-adapter:     at Timer.processTimers (timers.js:223:10)
> 2021-12-20 22:19:25.782 INFO   : zigbee2mqtt-adapter: 
> 2021-12-20 22:19:25.784 INFO   : zigbee2mqtt-adapter: z2m stdout: Zigbee2MQTT:info  2021-12-20 22:19:25: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{},"error":"SRSP - ZDO - mgmtPermitJoinReq after 6000ms","status":"error"}'
> ```
> 
> But most of the time there's simply... nothing.
> 
> I still have to try:
> 
> * extension cable
> * pairing different devices
> 
> I don't want to flash a different firmware, as for my purposes I'm looking for a cheap USB stick that I can recommend others to buy. It needs to be globally available, and work out of the box.

same for me, ordered the sonoff stick but IKEA bulbs aren't pairing. I didn't start debugging yet and will try to follow the points Koenkk advised.

**Comment by Hedda** [comment]
> **I don't want to flash a different firmware, as for my purposes I'm looking for a cheap USB stick that I can recommend others to buy.** It needs to be globally available, and work out of the box.

Guess it depends on your definition of "cheap" but regardless of the standard dilemma is that you can today not really find any Zigbee coordinator adapter hardware and Zigbee software solution that is fast/easy, cheap, and good: You can only pick two.

Would therefore strongly suggest that you slightly change your particular "works out-of-the-box" condition requirement for fast/easy when looking at any Zigbee Coordinator adapter as upgrading the firmware at some point will always be a minimum requirement if you want a stable Zigbee network. If not direct when you bought it (because you got lucky and received stock when the manufacturer just happen to have updated to a new revision) then later on once you discover bugs on there is a security exploit.

 highly recommend that you at least change the requirement to have a relatively fast/easy way to upgrade the Zigbee coordinator firmware on whatever adapter you choose. Suggest that you check out this "feature/ITead_Sonoff_Zigbee-delay" branch from JelmerT patch for cc2538-bsl which makes upgrading firmware much easier than following ITead's official guide (PR https://github.com/JelmerT/cc2538-bsl/pull/114 with patch for this sonoff plus dongle) -> https://github.com/JelmerT/cc2538-bsl/tree/feature/ITead_Sonoff_Zigbee-delay

Remember that all hardware today will need updates for firmware/drivers/software, so understand that by today's standard suggesting that users should never ever have to update their Zigbee coordinator firmware is almost like suggesting that users should be able to buy a smartphone and never update it either, (then thinking they can ask for support and not expecting being told to upgrade before getting any help). 

![image](https://user-images.githubusercontent.com/6320001/146933260-34c46411-edf8-4d0d-adc5-f822c7848d99.png)

**Comment by flatsiedatsie** [comment]
ON TOPIC: 

I just tried flashing the latest development firmware (`CC1352P2_CC2652P_launchpad_coordinator_20211217`) onto the Sonoff stick, using the excellent Python script recommended by @Hedda 

I still cannot pair my IKEA lightbulb.

- The Conbee stick is still unplugged from my other Raspberry Pi, so it's no interfering.
- I'm using an extension cable now.
- I tried keeping the Sonoff stick very close to the lightbulb as it was flashing.

I still don't see any reaction from Zigbee2MQTT.


OFF TOPIC:

> highly recommend that you at least change the requirement to have a relatively fast/easy way to upgrade the Zigbee coordinator firmware on whatever adapter you choose.

Yes that's definitely something I'd like too. I'd like to build firmware updating of the USB stick into the [Webthings Addon](https://github.com/kabbi/zigbee2mqtt-adapter) I work on. I've just updated that addon to add security support. I could add update flashing next.

> feature/ITead_Sonoff_Zigbee-delay

Very nice. If I understand correctly, USB sticks like Zig-Ah-Zig-Ah also have this feature.

This is my list of desired features for a USB Zigbee stick:
- Works with a wide range of devices, including IKEA buttons and sensors.
- Pairing is usually succesfull on the first try.
- Can handle enough devices for an average home and has enough range by itself.
- Can easily be purchased all over the world and at any time through 'big' retail channels.
- Likely comes with a working firmware at purchase time.
- Can be upgraded through Linux command.

Nice to have:
- Works with Zigbee2MQTT, but useful if the stick also has value for other smart home systems that might not build on Zigbee2MQTT
- Cheap

So far I've been recommending the Conbee2 stick. I've never once upgraded it since purchasing it years ago, but it seems to work fine so far.Theoretically it can be updated from the linux command line (which I'm attempting now, but I'm not having much luck so far)

The mayor downsides of the Conbee2 are:
- The Zigbee2MQTT project doesn't seem to reallty recommend it?
- It's relatively expensive.

TLDR: if the Sonoff stick could be made to work well, I think it's a winner.

**Comment by Hedda** [comment]
> > feature/ITead_Sonoff_Zigbee-delay
> 
> Very nice. If I understand correctly, USB sticks like Zig-Ah-Zig-Ah also have this feature.

@flatsiedatsie Yes such "Auto BSL" (formerly known as BootStrap Loader) feature allows you to trigger entering into bootloader mode via software and is built into many new microcontrollers, including CC2538, CC2652 and CC1352, but not supported by all: 

https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/README.md

Auto BSL is however not a feature of CC2530 and CC2531, however for those you can manually force it to enter bootloader and then flash them with zigpy-znp tools -> https://github.com/zigpy/zigpy-znp/blob/dev/TOOLS.md#flash-operations


> **I'd like to build firmware updating of the USB stick into the [Webthings Addon](https://github.com/kabbi/zigbee2mqtt-adapter) I work o**n. I've just updated that addon to add security support. **I could add update flashing next.**

Off-topic; Then suggest you check out and take inspiration from ESP Web Tools https://esphome.github.io/esp-web-tools/ (which support OTW flashing of firmware via a web browser using Web Serial API, though currently only for ESP based MCU chips), as well a https://github.com/electrolama/llama-bsl (fork of cc2538-bs with some added auto-detection feature and more similar features planned to make OTW flashing of firmware more user-friendly)

> This is my list of desired features for a USB Zigbee stick:
> 
> * Works with a wide range of devices, including IKEA buttons and sensors.
> * Pairing is usually succesfull on the first try.
> * Can handle enough devices for an average home and has enough range by itself.
> * Can easily be purchased all over the world and at any time through 'big' retail channels.
> * Likely comes with a working firmware at purchase time.
> * Can be upgraded through Linux command.

You need to remember that most of the things on your list do not depend on the Zigbee hardware adapter or even the Zigbee stack firmware, instead, those things will rather depend on the Zigbee software solution and the Zigbee devices that you connect. Understand that Zigbee is the wild wild west as not all manufacturers do not strictly follow the rules, and all of the rules are not even strict rules but more like guidelines. So implementing support for different Zigbee devices from hundreds or thousands of manufacturers is hard and complex. Jeedom developers recently wrote a great blog post article which makes for a good read summing all that up better than I can:

https://blog.jeedom.com/6107-parlons-zigbee/

A couple of quick tips for any new Zigbee user are; upgrade Zigbee coordinator firmware before beginning. Stay away from Aqara, Xiaomi, as well as Tuya devices. When starting to add devices begin by adding at least a few main-powered devices that will work as Zigbee router (a.k.a. repeater) devices, preferably flash CC2652 and/or CC1352 adapters with Zigbee router firmware and use those as known working Zigbee routers to create a stable network mesh at its core.

> The mayor downsides of the Conbee2 are:
> 
> * The Zigbee2MQTT project doesn't seem to reallty recommend it?
> * It's relatively expensive.

For now, recommend using Texas Instruments based adapters with Zigbee2MQTT. ConBee/RaspBee support is only experimental is Zigbee2MQTT. So is Silicon Labs (Silabs EZSP) based adapters.

https://www.zigbee2mqtt.io/information/supported_adapters.html

Texas Instruments Z-Stack 3 based asapters also work with Home Assistant's ZHA and Jeedom's Zigbee plugin

ConBee/RaspBee is currently really only good if used with dresden elektronik's own deCONZ software and not third-party.

**Comment by flatsiedatsie** [comment]
Update: success! I was just able to pair an IKEA bulb with the Sonoff 3.0 Plus stick running the latest development coordinator firmware. It even paired on the first attempt this time.

Turns out there was an error in my code, and because of that the security wasn't being implemented. This meant that the new Raspberry Pi was still using the default `PAN_ID` and `Network_key`. 

I suspect that the IKEA bulb didn't see the need to pair with the same controller it was technically already paired too.

**Comment by nicoolaro** [comment]
> Update: success! I was just able to pair an IKEA bulb with the Sonoff 3.0 Plus stick running the latest development coordinator firmware. It even paired on the first attempt this time.
> 
> Turns out there was an error in my code, and because of that the security wasn't being implemented. This meant that the new Raspberry Pi was still using the default `PAN_ID` and `Network_key`.
> 
> I suspect that the IKEA bulb didn't see the need to pair with the same controller it was technically already paired too.

care to explain details i am having same problem with Ikea

**Comment by flatsiedatsie** [comment]
@nicoolaro I'm not sure what to say. I think it all worked as normal after I flashed the beta firmware to the Sonoff stick. In my case I had two conflicting controllers with the same PAN_ID, which I believe is what was causing my issues.

**Comment by Hedda** [comment]
Update! Z-Stack 3.x.0 20211217 firmware release image has now been moved to Koenkk's master and should be seen as stable:

https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator/Z-Stack_3.x.0/bin

Again, recommend patching using JelmerT 's "ITead_Sonoff_Zigbee-delay" branch from this pull request

https://github.com/JelmerT/cc2538-bsl/pull/114

Hopefully ITead will begin pre-flashing this release to future batches, but keep upgrading will still always be recommended.

**Comment by flatsiedatsie** [comment]
I bought another Sonoff dongle to test. 

Out of the box it did connect to IKEA devices this time. This conforms that my previous issues were causes by having two clashing PAN ID's.

**Comment by TeslaCLT** [comment]
I also have just receive the Sonoff Zigbee 3.0 USB Dongle Plus from ITead. It is running the zStack3x0 firmware at level 20210708. As others here saw, I have had tons of pairing problems. This is a fresh install, so no existing data being migrated over (I'm coming off SmartThings). 

My first Ikea bulb paired right away without issue, but after none of my Ikeas, Sengleds, or eWeLights pair normal and fail during the interview. However, if I shut down z2m and run herdsman in debug mode via "DEBUG=zigbee-herdsman* npm start", they all pair without issue on the first try (from my perspective, one device did have to have multiple attempts per the log).

I'm not comfortable yet with flashing to another firmware, but as I get better acquainted with the coordinator, I'll give it a shot. I just wanted to share how debug mode helped me in case it helps other not rip their hair out!

**Comment by flatsiedatsie** [comment]
I just did another test.

- Current development version of ZIgbee2MQTT
- Out-of-the-box Sonoff stick (unflashed). Don't know which version it's running (how can I tell?)

Before I disabled debugging I was able to pair an IKEA motion sensor.

- Per @TeslaCLT I disabled debugging (which was enabled in my previous tests)

The bulb connected normally and I can toggle it and change its brightness.

I then tried to connect a second IKEA bulb of the same type, and that also connected fine.

**Comment by notenoughtech** [comment]
If you open z2m log in the terminal after restarting the service you'll get
the coordinator data including the firmware

**Comment by flatsiedatsie** [comment]
@notenoughtech thanks, found it. 

`"coordinator":{"ieee_address":"0x00124b0024XXXXXX","meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20210708,"transportrev":2},"type":"zStack3x0"}`

**Comment by TeslaCLT** [comment]
@flatsiedatsie I actually can only get things to pair if I put herdsman in debug mode (beyond that first bulb) -- but perhaps that is what you meant and misspoke? The command I provided (from another thread) turns on debug mode for herdsman.

Right now I have my 30 devices paired with the last 5 still on SmartThings. I'm going to wait a week or so to make sure the network is stable. I've noticed that my map currently has everything paired directly with the coordinator, though with sibling links to other routers. Of the 30 devices, only 1 isn't a router, but I'm still surprised to see none of them pairing with the other routers around them.

**Comment by flatsiedatsie** [comment]
> I actually can only get things to pair if I put herdsman in debug mode

Yes I understood. That's why I did a test here _without_ debug mode. The fact that is still worked for me without debug mode, though being on the _dev_ version of Z2M, might indicate that the issue has already been solved in a newer version of Z2M?

Then again, it might be something else.

> I'm still surprised to see none of them pairing with the other routers around them.

Interesting. My network graph usually shows a lot of routing over secondary devices (IKEA lightbulbs mostly).

**Comment by TeslaCLT** [comment]
Oh, you mean the dev branch of z2m. I misunderstood/misread and thought you meant the dev branch of the Sonoff firmware. I may indeed look into the dev branch of z2m then. 

FYI, a day later, I still have the same pairing issues. I have to turn on debug mode to pair anything. So hopefully, as you said, the newer release will clear things up.

**Comment by flatsiedatsie** [comment]
Ah, sorry for the confusion.

**Comment by oferwald** [comment]
Same issue here, latest firmware (1217). Pairing is really hard, and it seems nearly impossible via a router. (I was able to pair a few devices directly to the coordinator, after some hard resets, but this is not the way). I could not pair devices from sonoff itself, such as a motion detector from the SNZB series.
When activating debug mode and trying to pair, it seemed like there was a storm of messages, but the interview didn't even start. And I am with the latest-dev branch.
@Koenkk I do have a CC2531 sniffer, what capture would you like, and what scenarios do you need me to sniff? When allow_join is true? when I enable a specific router to allow_join? I don't want to provide with useless data, and since the network here has 46 devices that were already paired. there is a lot of useless data in my captures. LMK.

**Comment by u8915055** [comment]
Just to add to this thread.. and full disclaimer, i am using ZHA now but i wanted to share this because i am seeing a difference under 1217. I hope to go back to Z2M soon but was having too many issues.  With ZHA  I was able to pair about 25 sensors to it.. various ones from Sonoff, Aqara, Konke, etc.  I have lots of sensors.  I was able to get every single sensor to pair.  some of them i had to pair a couple times to get the battery state to come up properly.. been stable ever since.  However i will say the Sonoff ones are the hardest to pair.  Took me several tries to get them to pair but eventually they do.  I am finding aqara sensors especially pairing better with 1217.  Overwhelmingly i find the tuya sensors pair the best.

**Comment by Koenkk** [comment]
@oferwald is your network fresh or did you migrate from a CC2531 to the sonoff stick without repairing devices?

**Comment by oferwald** [comment]
@Koenkk Migrated from the CC2531 without repairing devices, I did have to power off some devices that allowed them to connect with the new coordinator.

**Comment by Koenkk** [comment]
@oferwald this is probably the problem, see e.g. https://github.com/Koenkk/zigbee2mqtt/issues/8670. If you are experiencing pairing problems after migrating from a CC2530/CC2531 to a sonoff stick without repairing devices it is recommended to repair a few routers starting from those who are the closest to the coordinator.

**Comment by oferwald** [comment]
@Koenkk Ok, I will do that, any special re-pairing instructions? should I force them out of the network before re-pairing? Should I change any network params? change the keys or something?

This still does not explain why I am unable to pair the sonoff motion sensor when I am near the coordinator, or is it connected somehow? maybe because I originally tried to pair it near some routers?

Thanks for your patience

**Comment by Koenkk** [comment]
@oferwald you can repair them by just enabling joining in z2m and factory resetting the routers. Sonoff motion sensors are always hard to pair (even with a CC2531) so these are not the best reference.

**Comment by oferwald** [comment]
@Koenkk Hi, I tried factory reset for some tuya covers acting as routers, this is what I get:
Zigbee2MQTT:warn  2022-01-09 12:52:09: Device 'Living room diagonal shutter' left the network
dress":"0xbc33acfffe682925"},"type":"device_leave"}'

The device is not interviewing again, and when I take the power out and do a reset again, I get the same message again (although it has already left the network? how can it leave again?)

And now those few routers are out of my network, not joining again. :( something is really off here. any ideas?

**Comment by Koenkk** [comment]
@oferwald probably those routers try to join again via other routers where pairing is not working with.
- Try to disable joining and permit it again via the coordinator; send to `zigbee2mqtt/bridge/request/permit_join` payload `{"value": true, "device": "coordinator"}`and factory reset the router.
- If that doesn't help, power off all routers in the neighbour and factory reset the router.

**Comment by oferwald** [comment]
@Koenkk Thanks for the advice, I tried the disable joining and allowing only coordinator to join, but to no avail.
I can't take every router down now, because this means taking electricity out in most of my house.
The device keeps "leaving" the network, when I tried a force remove, it said it is not even there, I truly wonder how is it even possible for a device that left to leave again.

I did try deleting the entire docker config directory, this didn't help at all. :(

My next plan is to connect the old C2531 back, allow devices to join, and replace it again, but I will probably only attend to this tomorrow. Some thing is badly broken, maybe I did something, nothing that I am aware of.

I will be happy to provide any debug information that might help. And again, thanks for trying to help.

**Comment by Koenkk** [comment]
@oferwald I'm afraid that this cannot be fixed from the z2m side. The problem is that the CC2531 doesn't remember all devices (only 5 since that is the max), these do not end up in the coordinator_backup.json and my expectation is that only devices which are in coordiantor_backup.json can be paired via.

**Comment by oferwald** [comment]
@Koenkk I am not sure that I follow you, the current coordinator_backup.json I have only includes 3 devices, all of which were paired directly with the new coordinator, and none of which are a router at all (all are sensors). What will happen if I manually add devices to this file? (how to find a key?) What will happen if I delete this file?

**Comment by Koenkk** [comment]
> What will happen if I manually add devices to this file? (how to find a key?)

This won't work, the key is generated when the device joins the network. 

> What will happen if I delete this file?

nothing, z2m will just regenerate it on next shutdown

**Comment by oferwald** [comment]
So if I understand correctly, I have a coordinator that is not connected to any router, yet still able to coordinate the network, and this is the reason I can't pair almost anything new, and this is not solvable by z2m.
For the time being, I will use my hack to extend the network, its not like I have new devices everyday. But I assume that as this dongle will become more widespread, the issue will become quite common.

**Comment by Koenkk** [comment]
>  I have a coordinator that is not connected to any router, yet still able to coordinate the network,

Zigbee doesn't work in such way, devices can communicate with each other when they use the same pan_id, ext_pan_id, channel and network key. So they can communicate, but the coordinator refuses to allow any devices to join since the device is not known in the coordinator.

> But I assume that as this dongle will become more widespread, the issue will become quite common.

The issue is not with the dongle itself but with the migration from cc2531 -> sonoff. The cc2531 backup doesn't contain all the devices which is the problem (due to limits of the CC2531). It can be fixed by either repairing all routers or starting from scratch

**Comment by castorw** [comment]
@oferwald Could you please provide the following:
* NV dump of the original dongle (use https://github.com/zigpy/zigpy-znp/blob/dev/zigpy_znp/tools/nvram_read.py)
* NV dump of the new coordinator (use https://github.com/zigpy/zigpy-znp/blob/dev/zigpy_znp/tools/nvram_read.py)
* `coordinator_backup.json`

There is a variety of variables in the backup and restore process. Most likely the issue is not isolated to a specific dongle but rather firmware or firmware combinations.

**Comment by Hedda** [comment]
> I'm not comfortable yet with flashing to another firmware, but as I get better acquainted with the coordinator, I'll give it a shot.

Tip for upgrading firmware without the dongle enclosure is to follow https://github.com/JelmerT/cc2538-bsl/pull/114 (recommend backup NVRAM first)

**Comment by osos** [comment]
> 
> The issue is not with the dongle itself but with the migration from cc2531 -> sonoff. The cc2531 backup doesn't contain all the devices which is the problem (due to limits of the CC2531). It can be fixed by either repairing all routers or starting from scratch

@Koenkk, do you recommend that all routers are repaired also while migrating from CC2531 -> sonoff, I think that is different from the docs: https://www.zigbee2mqtt.io/guide/faq/#what-does-and-does-not-require-repairing-of-all-devices

I have a few IKEA bulbs which re hard to get access to, and they seem to work. Most troubles are with the remotes on battery (difficult to pair and seem to drain battery).

What should be the best procedure to re-pair the routers?

**Comment by Koenkk** [comment]
> do you recommend that all routers are repaired also while migrating from CC2531 -> sonoff

Lets see what @castorw finds, it would help if you provide the required logging: https://github.com/Koenkk/zigbee2mqtt/issues/9117#issuecomment-1008742890

**Comment by notenoughtech** [comment]
I finally had some time to test the latest firmware, thank you Koen and everyone who has contributed to this. 
TLDR: 3 days in I have no issues and one instance (*fixed) of sensor pairing but not reporting.
More detailed impressions of the master and development branch impressions here:
https://notenoughtech.com/home-automation/is-sonoff-zigbee-dongle-plus-finally-ready-for-launch/

**Comment by osos** [comment]
@notenoughtech thank you for the update. Regarding the flashing of the Sonoff dongle you should have a look at https://github.com/JelmerT/cc2538-bsl/pull/114 with the pull-request in there flashing the dongle is really easy.

**Comment by notenoughtech** [comment]
I covered this in my vid/article.
While convenient for anyone working on the firmware, pressing the button is
hardly a chore when I do it once every 6 months or so :)
        Mat Zolnierczyk

Maker, Robotics Engineer

Owner | NotEnoughTech
***@***.***
notenoughtech.com
Teesside, North Yorkshire, UK
[image: facebook] <https://www.facebook.com/NotEnoughTECH/>
[image: twitter] <https://twitter.com/NotEnoughTECH>
[image: linkedin] <https://www.linkedin.com/in/mat-zolnierczyk/>
[image: instagram] <https://www.instagram.com/notenoughtech/>

**Comment by flatsiedatsie** [comment]
> While convenient for anyone working on the firmware, pressing the button is hardly a chore when I do it once every 6 months or so

I think you overestimate my mother's skills :-)

**Comment by notenoughtech** [comment]
> I think you overestimate my mother's skills :-)

Shouldn't you explain how dangerous the use of technology is? haha! That's fair play and kudos to her sense of adventure!

**Comment by Hedda** [comment]
> While convenient for anyone working on the firmware, pressing the button is hardly a chore when I do it once every 6 months or so :)

Surly it's a chore to unscrew the enclosure even once just to upgrade the firmware when the Auto-BSL feature makes it so easy ;)

Not all feature "Auto-BSL" https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/README.md

**Comment by notenoughtech** [comment]
What do you mean?
I have a hole in mine now. 😁😁😁
Hardware fix to a software problem 😉😉😉

**Comment by Hedda** [comment]
By the way, someone now claim that Auto-BSL also works with Sonoff in the ZigStar GW Multi tool too -> https://github.com/xyzroe/ZigStarGW-MT/issues/2

**Comment by notenoughtech** [comment]
I have 5 decent coordinators now. 
Any ideas for fair tests that don't require RF equipment? 
I can use cc2531 for sniffing while testing but that's about it

**Comment by flatsiedatsie** [comment]
- Log the signal strength in a smart home system of a fee devices around the home, and see how that value changes with the different coordinators? I recommend the Webthings Gateway (I'm updating the Zigbee2MQTT addon soon, lots of nice new features).

**Comment by notenoughtech** [comment]
That's kinda what I've done in my original article:
![image](https://user-images.githubusercontent.com/42407403/148968733-5938f93c-14cb-40ee-8223-2e15ffca7e36.png)

**Comment by Koenkk** [comment]
Please keep this issue on topic, otherwise it gets harder to track this. If you want to have a discussion please create a discussion for it: https://github.com/Koenkk/zigbee2mqtt/discussions

**Comment by flatsiedatsie** [comment]
Back on topic: are people seeing any issues from having it plugged directly into their Raspberry Pi? Mine is working fine directly in a Pi 3, but I haven't tested it with a Pi 4 yet (which are said to cause more interference).

**Comment by migube** [comment]
usb 2 or 3 ports on pi4 might make a difference, or behind usb2.0 hub

**Comment by TeslaCLT** [comment]
> Oh, you mean the dev branch of z2m. I misunderstood/misread and thought you meant the dev branch of the Sonoff firmware. I may indeed look into the dev branch of z2m then.
> 
> FYI, a day later, I still have the same pairing issues. I have to turn on debug mode to pair anything. So hopefully, as you said, the newer release will clear things up.

Given the last few posts, I should mention that in my issue (requiring herdsman in debug mode to successfully pair devices), I have my Sonoff dongle plugged into a usb2 port on my thin client. Should I plug it into a usb3 port instead? I didn’t think throughput would need to be higher than usb2, but perhaps there are other considerations to make?

**Comment by Hedda** [comment]
> Back on topic: are people seeing any issues from having it plugged directly into their Raspberry Pi? Mine is working fine directly in a Pi 3, but I haven't tested it with a Pi 4 yet (which are said to cause more interference).

Yes any USB 3.0 port is known to cause interference with all Zigbee adapters because it is also operating on 2.4GHz frequency, and the same problem can be seen with Wi-Fi USB adapters and Bluetooth USB adapters, so not only with Zigbee adapters. 

And there is not only an issue with plugging a Zigbee dongle directly into the USB 3.0 port itself but also leaving a Zigbee dongle close to other unshielded cables/wires or unshielded devices that are connected to USB 3.0 ports, (especially infamous are connecting USB 3.0 SSD or harddrives to USB-port and leaving those cables/wires or device close to the Zigbee dongle antenna).

https://www.reddit.com/r/technology/comments/136g7y/usb_30_has_been_found_to_cause_interference_that/

This is not only an issue with Raspberry Pi 4 but really on all type of computers with USB 3.0 ports. See report on usb.org by Intel:

https://www.usb.org/sites/default/files/327216.pdf

Therefor the general recommendation is to use a USB 2.0 port or if use USB 3.0 port then plug in a USB 2.0 hub first and connect the Zigbee adapter into the USB 2.0 hub instead.

**Comment by castorw** [comment]
@oferwald I have reviewed the attached files:
* Did you pair any extra devices to the new coordinator - I see no correlation in devices between the backups. However there are som extra devices with APS link keys.
* The original coordinator was running Z-Stack 1.2 - therefore no APS link keys could have been negotiated before.
* Are you experiencing pairing problems after re-pairing all routers?

Theory that comes in mind is that after upgrade from Z-Stack 1.2 to 3/4 the coordinator tries to negotiate APS link keys and screws up somehow. I will need to do some more testing on this matter.

**Comment by Hedda** [comment]
> @oferwald I have reviewed the attached files:
> 
> * Did you pair any extra devices to the new coordinator - I see no correlation in devices between the backups. However there are som extra devices with APS link keys.
> * The original coordinator was running Z-Stack 1.2 - therefore no APS link keys could have been negotiated before.
> * Are you experiencing pairing problems after re-pairing all routers?
> 
> Theory that comes in mind is that after upgrade from Z-Stack 1.2 to 3/4 the coordinator tries to negotiate APS link keys and screws up somehow. I will need to do some more testing on this matter.

FYI, zigpy-znp developers also seen some issues when backup Z-Stack Home 1.2 dongle and restored to Z-Stack 3.0, see ex:

https://github.com/zigpy/zigpy-znp/pull/92

https://github.com/zigpy/zigpy-znp/issues/70

**Comment by oferwald** [comment]
@castorw I paired exactly three devices to the new coordinator, two water leak sensors, and a temp sensor that stopped working with the previous coordinator.

The CC2531 was naturally 1.2, the source routing version

For the time being, I didn't re-pair any routers, I have something like 25 routers, most of them are switches and cover switches, and taking them out is only by taking some power off my house. And I am not sure that I know exactly how to force a reset for each of them. I did try to pair a new router about 2cm from the new coordinator and no pairing happened at all (no interview), so I waited with this more tedious process for your inputs, mainly in the hope this process won't be needed. :)

Let me know if I can provide any further information, or if you want me to test anything

thanks for your help!

**Comment by castorw** [comment]
@Koenkk @oferwald @Hedda I have confirmed that ZH's current backup restore mechanism does indeed create this corruption as described in https://github.com/zigpy/zigpy-znp/pull/92. I have fixed the data structure locally and will get to creating a full fix PR soon (a few days at most).

The fix also needs to address problems in current setups where the migration corrupted the address manager table, so there will be an address manager table check on startup which will fix the table entries if necessary.

I will keep this issue updated as well.

**Comment by marcelrv** [comment]
I had great difficulties paiting Xiaomi Aquera devices with the stock firmware.
I finally managed to pair the devices with the development branch version: CC1352P2_CC2652P_launchpad_coordinator_20220103.zip

So far seems to be working fine. Pairing took a lot of time, and only finished after 20 or so times clicking the buttons shortly to finish the process. Will continue monitoring if anything locks up after a while.

**Comment by notenoughtech** [comment]
Did you have any log errors? I found sonoff zb temp sensors giving me bind
timeouts. To successfully pair these I had to turn join off, reset sensor
and wait 2min. Then pair it again. For some reason it helps to issue join
true command on stubborn sensors (I'm not sure what's the logic behind is
or a sheer coincidence but as soon as I trigger join again ) it completes
the interview process.

**Comment by Hedda** [comment]
> Tip FYI, zigpy-znp developers also seen some issues when backup Z-Stack Home 1.2 dongle and restored to Z-Stack 3.0, see:
> 
> [zigpy/zigpy-znp#92](https://github.com/zigpy/zigpy-znp/pull/92)
> 
> and
> 
> [zigpy/zigpy-znp#70](https://github.com/zigpy/zigpy-znp/issues/70)

@castorw another tip could be to also check out this other maybe related issue and fix -> https://github.com/zigpy/zigpy-znp/pull/120

**Comment by castorw** [comment]
I have created a PR addressing this issue (https://github.com/Koenkk/zigbee-herdsman/pull/495). @Koenkk @oferwald Can you please test if it fixes the issue before I finish the PR by updating and adding tests?

**Comment by oferwald** [comment]
Hi @castorw,

Can you provide me with instructions on how to test this? I am using the latest-dev docker

Thanks

**Comment by castorw** [comment]
> Hi @castorw,
> 
> 
> 
> Can you provide me with instructions on how to test this? I am using the latest-dev docker
> 
> 
> 
> Thanks

@Koenkk could you please provide instructions?

**Comment by Koenkk** [comment]
For Docker
```bash
docker exec -it zigbee2mqtt sh # replace zigbee2mqtt with the container name
cd node_modules
rm -rf zigbee-herdsman
apk add make gcc g++ python3 linux-headers git
git clone https://github.com/castorw/zigbee-herdsman.git -b znp-fix-addrmgr-empty
cd zigbee-herdsman
npm ci
npm run build
exit
docker restart zigbee2mqtt # replace zigbee2mqtt with the container name
```

**Comment by oferwald** [comment]
Thanks for the instructions,

I guess I am running the following version now:
Zigbee2MQTT:info  2022-01-18 18:36:11: Starting Zigbee2MQTT version 1.22.2 (commit #414c51f)

Anyhow, there is no change to coordinator backup, what would you like me to do (except trying to pair new devices or attempt to re-pair an existing one? need any log info?)

LMK

**Comment by castorw** [comment]
@oferwald What would be great if you were able to run Z2M with `DEBUG=zigbee-herdsman:adapter:zStack:startup*` so we can verify it works properly - you should see message stating `(#9117) verifying address manager table for post-migration corruption`. And then another one saying it either fixed or didn't fix AMT entries.

Afterwards try pairing a device and check if it works. If it does not, please try unplugging the adapter and restarting Z2M and please let me know of the steps you took and whether it works.

Thanks.

**Comment by oferwald** [comment]
@castorw 

No such line in my logs, I did start with debug and have tons of output though:
debug 2022-01-18 20:06:57: Loaded state from file /app/data/state.json
info  2022-01-18 20:06:57: Logging to console and directory: '/app/data/log/2022-01-18.20-06-56' filename: log.txt
debug 2022-01-18 20:06:57: Removing old log directory '/app/data/log/2022-01-09.17-22-16'
info  2022-01-18 20:06:57: Starting Zigbee2MQTT version 1.22.2 (commit #414c51f)
info  2022-01-18 20:06:57: Starting zigbee-herdsman (0.13.188)
debug 2022-01-18 20:06:57: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/app/data/coordinator_backup.json","dat
abaseBackupPath":"/app/data/database.db.backup","databasePath":"/app/data/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"
HIDDEN","panID":6754},"serialPort":{"path":"/dev/ttyUSB1"}}'
info  2022-01-18 20:06:58: zigbee-herdsman started (resumed)
info  2022-01-18 20:06:58: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20211217,"transportrev":2},"type":"zStack3x0"}'
debug 2022-01-18 20:06:58: Zigbee network parameters: {"channel":11,"extendedPanID":"<redacted>","panID":6754}

--------------
Waiting for further instructions

**Comment by castorw** [comment]
@oferwald 
This output does not seem to include DEBUG output. Please try and refer to this https://www.zigbee2mqtt.io/guide/usage/debug.html#enabling-logging.

**Comment by oferwald** [comment]
Hi @castorw 

I thought that the lines starting with debug were indicating that it is using debug,
It was started with the following command:
docker run -it --name=zigbee2mqtt -v /root/docker/zigbee2mqtt:/app/data --device=/dev/ttyUSB1 -e TZ=Asia/Jerusalem -v /run/udev:/run/udev:ro --privileged=true -e DEBUG=zigbee-herdsman:adapter:zStack:startup* koenkk/zigbee2mqtt:latest

Also, the log file size is about 30x what it previously was. including many starting with "debug"

I will be glad to provide you with the full log if needed, but I guess is that something is still a miss.

**Comment by castorw** [comment]
Hi @oferwald, please provide the log or at least lookup the line I referenced above so we can confirm you are running the dev ZH from my repo :-)

To the topic of logging, there are 2 logging mechanisms:
* general logger, which has the standard severities (debug, info, warn, error)
* specific development debug logging which needs to be enabled explicitly with the `DEBUG` environment variable

**Comment by oferwald** [comment]
Hello again, @castorw 

I was to blame for some, as I killed the docker and started it again, which made all the changes go away instead of restarting it, but I did it again, and now the log shows:
debug 2022-01-19 15:27:02: Loaded state from file /app/data/state.json
info  2022-01-19 15:27:02: Logging to console and directory: '/app/data/log/2022-01-19.15-27-01' filename: log.txt
debug 2022-01-19 15:27:02: Removing old log directory '/app/data/log/2022-01-10.17-40-26'
info  2022-01-19 15:27:02: Starting Zigbee2MQTT version 1.22.2 (commit #414c51f)
info  2022-01-19 15:27:02: Starting zigbee-herdsman (0.13.194)

(notice the changed zignee-herdsman version)
However, this is still not working, nothing that I see in my log, I did get to shell and your code is there, in fixup.js and some other locations

If there is a way to provide you with the full log privately, just let me know

**Comment by castorw** [comment]
@oferwald Okay so lets presume it worked okay. Can you please provide me with NV dump like you did before - that way I can confirm the fix did what it was supposed to. Also did you try pairing devices to see if the problem has been resolved?

**Comment by oferwald** [comment]
@castorw file is attached, will try some re-pairing too and will report back in a few minutes
[files2.tar.gz](https://github.com/Koenkk/zigbee2mqtt/files/7897623/files2.tar.gz)

**Comment by castorw** [comment]
@oferwald According to the dump the AMT is fixed, so please try if it works ;-)

Also I kinda omitted the last sentence in your previous message - you can find me on Telegram (@castorko).

**Comment by castorw** [comment]
@Koenkk @oferwald It seems this fix didn't completely resolve this issue. I will be performing some more tests on this. However I suspect there is similar problem with the TCLK table and therefore joining a ZB3 device might fail due to inability to store APS link key information which is required by spec R21+.

I will test this more thoroughly in the upcoming days. Thank you for your patience.

**Comment by oferwald** [comment]
Hi,

I want to thank @castorw for his amazing help in fixing this issue for me, I am writing the cause and process that was used that enabled me to pair devices again, having learned some more on zigbee and stuff. :)

NOTICE - This worked for me, YMMV!!! kids, don't do this at home unless you know what you are doing, it is probably easier to re-pair all routers (not in my case, with 30 routers that some are inside walls, when pairing seemed off)

The root cause was apparently an EPID conflict, according to @castorw some CC2531 versions used "extended_pan_id": "dddddddddddddddd" instead of the ieee address of the device, and when a new device was trying to pair the coordinator and router were conflicting.

so, how do you fix this?
shutdownz2m
clear the nvram of your current coordinator with znp nvram_reset, for me it was:
`python zigpy_znp/tools/nvram_reset.py  /dev/ttyUSB1`

now edit your coordinator_backup.json
add the line
`"extended_pan_id": "dddddddddddddddd",`

for me it was just below "pan_id"

restart your z2m

live happily ever after (maybe)

I assume that a better fix will be provided in some near future. 👍

**Comment by l0rda** [comment]
> I had great difficulties paiting Xiaomi Aquera devices with the stock firmware. Also when pairing was done half, or in another app, the only way to re-initiate the pairing seemed to be to reflash the device with new mac
> 
> I finally managed to pair the devices with the development branch version: CC1352P2_CC2652P_launchpad_coordinator_20220103.zip
> 
> So far seems to be working fine. Pairing took a lot of time, and only finished after 20 or so times clicking the buttons shortly to finish the process. Will continue monitoring if anything locks up after a while.

i confirm, successfully paired 5 xiaomi sensors with this stick firmware, but only after 20-30 times. I still cannot pair ikea bulbs.

**Comment by danneh82** [comment]
I seem to be having the same issue but not 100% sure.

I migrated from a cc2531 to the zigbee3.0 following the below guide to copy the IEEE
![image](https://user-images.githubusercontent.com/67841514/151662561-2c473ffd-ae9b-41d1-b7d1-116d788e5e4b.png)

I had a few ikea switches batteries die which wouldn't pair again. 

I've flashed the sniffer firmware on a spare cc2531. When putting the switch into pairing mode i see these errors:-

![image](https://user-images.githubusercontent.com/67841514/151662387-d3a70d94-e974-450b-acc6-fc51d683ffbb.png)

Checking Settings => raw and the coordinator backup the EPID is matching the coordinator IEEE
![image](https://user-images.githubusercontent.com/67841514/151662443-ce36c23f-38ee-4108-9706-f1645c22210f.png)

![image](https://user-images.githubusercontent.com/67841514/151662461-a5fe8512-c562-4b7b-ba4b-658bc63cf1f6.png)

> oferwald 
so, how do you fix this?
shutdownz2m
clear the nvram of your current coordinator with znp nvram_reset, for me it was:
python zigpy_znp/tools/nvram_reset.py /dev/ttyUSB1
now edit your coordinator_backup.json
add the line
"extended_pan_id": "dddddddddddddddd",
restart your z2m

for the above fix can i do that from within home assistant or do i need to plug the dongle into a pc?

Does the dongle then pick up the EPID from the backup file and right it to nvram?

**Comment by oferwald** [comment]
The dongle should pick up the EPID from the backup file after you reset its nvram.

I ended up resetting most of my routers on the network anyhow, which was a huge pita, but things seem to work better now.

I believe that @castorw is probably working on a better fix for this.

**Comment by SkarfyFPV** [comment]
Hi there, I am having the same problem and trying to apply this fix, but I am new to all of this and I cannot find the way (or the shell) to run this command: python zigpy_znp/tools/nvram_reset.py /dev/ttyUSB1. 
I did the Home Assistant installation following the official doc: https://www.home-assistant.io/installation/raspberrypi
but I didn't install Home Assistant Container... 
Please any help will be appreciated.

**Comment by SkarfyFPV** [comment]
How can I delete the NVRAM without access to the shell? Or how can I access the shell? Thanks

**Comment by oferwald** [comment]
@SkarfyFPV My suggestion is that you just take the stick out to any computer that you can install zigpy on and have shell access to.

**Comment by SkarfyFPV** [comment]
Thank you @oferwald, this is a good idea. Would you recommend using docker containers instead next time a start from scratch?

**Comment by oferwald** [comment]
@SkarfyFPV I am not sure I am in a position to recommend anything, I am a simple user of this great project.
I have z2m and ha both running on docker, it was probably the simplest way for me personally to get going. It has its advantages and disadvantages (mainly for HA, as I am missing the controller). But I am ok with my current setup. It mostly works :). I don't think that there is really a one solution that fits everyone, so my suggestion is to just try and find out what is best for you.

**Comment by SkarfyFPV** [comment]
Thank you @oferwald I really appreciate your help. I have been some time thinking about trying with docker and I think I will just get another SD and try it. Also, I see a great advantage with docker, as you can have a few containers and just run the ones you need, which seems quite good to me for quick testing. Thanks again!

**Comment by danneh82** [comment]
Finally managed to reset the nvram and changed the EPID in the coordinator backup file. Happy to report that i have been able to get devices to pair again.

Almost completely back up and running bar one stubborn light which is refusing to go in pairing mode!

**Comment by drogfild** [comment]
@danneh82 Just to make sure (before I'm doing the same), you edited EPID in the backup file to be "dddddddddddddddd"? Previous it was same as your coordinator IEEE?

**Comment by danneh82** [comment]
> 

Yep exactly :)

**Comment by SkarfyFPV** [comment]
Hi! apologies, it took me more than expected to be able to test this. The result for me was unsuccessful, executed a few times :(

As I have directly installed HASS on my Pi I don't have access to the shell to run this commands, for this reason, I had to unplug my CC2652P and plug it into another Pi where I am running Debian to be able to clear the NVRAM. Maybe this is the reason for my failure. 

These are the steps I followed: 
1. shutdown z2mqtt
2. Unplug CC2652P from HASS and plug it into another Pi with Debian
3. Clear the NVRAM of the coordinator with znp nvram_reset, for me it was:
python zigpy_znp/tools/nvram_reset.py /dev/ttyUSB0
4. Edit my coordinator_backup.json
5. Modify the line with tag "extended_pan_id" to show as this: "extended_pan_id": "dddddddddddddddd",
6. Plug CC2652P into HASS Pi. 
7. Restart your z2mqtt

After these steps, my network was working with the already set devices but I could NOT pair anything else. I tried a couple of different bulbs and switches without any luck. 

Any other idea that comes to your mind which may help me to sort the problem? 

Thank you all!

**Comment by l0rda** [comment]
@SkarfyFPV i changed channel to 25 and it works as expected

**Comment by SkarfyFPV** [comment]
@l0rda if I change the channel then nothing works. None of the previously set devices are recognised.

**Comment by rvt** [comment]
Just my two cents if somebody finds this.., 
clearing the nvram only did not work for me but I did the below steps, but I am not sure what step really did the fix:

- stop zigbee2mqtt
- Do the nvram clear
- unplug the stick, and re-add (did you turn it off and on??)
- Change extended_pan_id to dddddddddddddddd
- in database.db, I removed all coordinators, for some reason I had multiple
- start zigbee2mqtt

After this all pairing did work directly, not even I had to keep items close to the coordinator.

This was after I upgraded from the good old CC2531 chip to the sonoff zigbee3 CC2652P stick.

**Comment by lordlala** [comment]
I am having the exact same issue as @SkarfyFPV .

All of a sudden my zigbee devices stop working and become unresponsive. If I try and remove and re-add, they never seem to repair.

What a trainwreck zigbee is :(

**Comment by Hedda** [comment]
There are still several issues/problems when upgrading from any CC2530/CC2531 adapter to any CC2652/CC1352 adapter.

If you have a "production environment" and therefore do not want to spend a lot of time troubleshooting to try to find the root causes then the general recommendation today is to just remove and re-repair all your mains-powered Zigbee Router devices, starting with the Zigbee Router devices closest to the Zigbee Coordinator adapter and then continue moving your aways outwards.

**Comment by rvt** [comment]
I wonder if there was somebody else that did set up a coordinators, perhaps somebody in the neighborhood... wel, at least in my DB I found coordinators that where still old, perhaps U used them once... after I removed them all and restarted zigbee2mqtt things started to work again

**Comment by hunterdrayman** [comment]
Just adding my experience of upgrading from CC2531 to CC2562P. I have all the woes mentioned here with devices not re-pairing. I removed all the routers and I do see a successful interview of a router. It appears in configuration.yaml and in the zigbee2mqtt device map but not in the front end device list. All of the sensors etc that were registered before the swap continue to talk to the new coordinator but no new ones can be added. This is after resetting the coordinator NVRAM etc.

Fortunately I took a VM snapshot of the host before replacing my trusty old CC2531 coordinator so it's now time to do a device swap and restore.  Amusingly, I sent back a recently purchased switch as being DOA - I wonder if I should admit my mistake to the vendor. Ooops!

**Comment by SkarfyFPV** [comment]
Thank you guys for chipping in. I honestly didn't find a solution for my problem and I am just waiting for a moment where I have time enough to build the HASS again from scratch... I did try all your suggestions but sadly none worked for me.

**Comment by Hedda** [comment]
Tip before troubleshooting further is to be sure to follow all the general advice and best practices to get a more optimal setup. 

## Improving Zigbee network range

Poor signal quality or interference can lead to transmission problems and show symptoms such as or errors when sending and/or receiving Zigbee messages that will cause significant degradation in Zigbee network performance. Below is some fundamental background information and tips for a Zigbee setup starting point to achieve better signal quality, coverage and range.

Following these essential optimization tips will typically resolve many known issues caused by a bad setup of Zigbee Coordinator adapter and Zigbee network mesh or Zigbee devices. At the very least taking these actions should improve most message transmitting and receiving issues caused by not knowing the basics needed to workaround related Zigbee limitations.

### Understanding common causes of problems in Zigbee setups

- Zigbee Coordinator adapter hardware:
  - Poor placement of the Zigbee Coordinator and/or wrong orientation of Zigbee Coordinator antenna.
  - Poor / outdated / incompatible Zigbee Coordinator firmware on the Zigbee Coordinator adapter.
  - Old, outdated or underpowered Zigbee Coordinator hardware or poor Zigbee Coordinator antenna.
- Zigbee Coordinator and Zigbee devices sensitive to all types of RFI/EMI/EMF interference:
  - Electromagnetic interference (EMI) from electronics caused by Electromagnetic Fields (EMF).
  - USB 3.0 ports and computer peripherals are known culprits of EMI/EMF disrupting Zigbee.
  - 2.4 GHz Radio Frequency Interference (RFI) from Wi-Fi Routers and Wi-Fi Access Points, or others.
  - Radio proximity as well conflicting radio frequencies (Wi-Fi) overlapping used Zigbee frequency.
- Too few or far distance between Zigbee devices in Zigbee network mesh:
  - Too many obstacles between the Zigbee Coordinator or Zigbee Router and Zigbee End Device.
    - Zigbee network mesh depends on Zigbee Router devices to extend range and coverage.
    - All individual Zigbee devices do have poor signal penetration and not long range.
  - Too few Zigbee Router devices Zigbee Coordinator or Zigbee Router and Zigbee End Device.
    - Only mains-powered devices can be repeaters that extend network range and coverage.
    - Any battery-operated products are a Zigbee End Device and will never be a Zigbee Router.

### Simple actions that should improve most Zigbee setups

- Zigbee Coordinator adapter hardware:
  - Try different physical placement and orientations of Zigbee Coordinator or its antenna.
    - Optimal placement of Zigbee Coordinator is as much in middle of the house as possible.
    - Try to place the Zigbee coordinator at some distance way from walls, ceilings and floors.
    - Try different orientations of your Zigbee antenna (or whole Zigbee coordinator adapter).
  - Update to a later released Zigbee Coordinator firmware on the Zigbee Coordinator adapter.
    - Updating FW is easy and community often provide newer firmware than manufacturer.
  - Buy a more powerful Zigbee Coordinator adapter based on a newer/modern Zigbee chip hardware.
    - Recommended as older adapter might work but have obsolete hardware and/or firmware.
- Zigbee Coordinator and Zigbee devices RFI sensitive as susceptible to all types of EMI/EMF interference:
  - Use a long USB extension cable to place Zigbee Coordinator away from interference and obstacles.
    - Try to make sure the USB extension cable has proper shielded (thick cables usually do).
    - Longer USB extension cable also make it easier to orient Zigbee Coordinator and antenna.
  - Only connect Zigbee Coordinator USB adapter to a USB 2.0 ports (or a powered USB 2.0 hub).
    - USB 3.0 ports and peripherals are infamously known for causing EMI/EMF interference.
  - Shield close unshielded computers and unshielded peripherals by using metal enclosures/chassis.
    - Single-board-computers and USB hard drives are especially known as source of EMI/EMF.
  - Change Wi-Fi Router and Access Points channels to not interfere with the default Zigbee channel.
    - Changing Wi-Fi channel on Wi-Fi devices is recommended over changing Zigbee channel.
  - Change Zigbee channel on Zigbee network if have too many strong sources of overlapping Wi-Fi.
    - Changing Zigbee channel is considered a last resort so make sure to do research before.
      - Normally require re-pairing all, and not all devices support all Zigbee channels.
- Too few or far distance between Zigbee devices in Zigbee network mesh:
  - Zigbee network mesh depends on having Zigbee Router devices to extend range and coverage:
    - Add additional mains-powered Zigbee devices known to be good as Zigbee Router device.
      - Add more Zigbee Router devices to extend network mesh range and coverage.
      - Note that not all mains-powered Zigbee devices act as a Zigbee Router device.
      - Some Zigbee Router devices are know to only work well with same brand devices.  
    - Buy a few known good dedicated Zigbee Router product (ex. “IKEA Tradfri Signal Repeater”).
      - Search community forums for "Zigbee signal repeater" or "Zigbee range extender".
    - Buy a few extra new Zigbee USB adapters to use after flashing with Zigbee Router firmware.
    - Reflash/reuse Zigbee USB adapters to Zigbee Router when upgrading Zigbee Coordinator.

## Best practices to avoid pairing/connection difficulties

Verify that you try to follow recommended best practices to avoid pairing and/or connection issues:

- If possible try to pair your Zigbee devices in their intended final location, (and not pair it next to the Zigbee coordinator and then need to move it after). 
  - Pairing a Zigbee device next to the Zigbee coordinator and then moving it later can result in dropped/lost connections or other issues. 
    - If the device you want to add is not brand new and as such never paired before then you always have to make sure to first manually reset the device to its factory default settings before you will be able to add/pair it. 
- Some battery-operated Zigbee devices are known to have problems with pairing if they have Low battery voltage. 
    - Some people have reported replacing the battery on their newly received Xiaomi/Aqara devices solved pairing issues. 
- Check that you have enough Zigbee router devices (also known as Zigbee signal repeaters or range extenders) and if you do not have any, invest and add some mains-powered devices that will work as Zigbee routers. 
    - Aim to start out with mains-powered devices before adding battery-operated devices as a "weak" Zigbee network mesh (e.g., the device is too far from the Zigbee coordinator or a Zigbee router) may prevent some devices from being paired. Zigbee router devices are also needed to increase the maximum of devices that can be connected to your Zigbee mesh network. 
    - Note that some Zigbee devices are not fully compatible with all brands of Zigbee router devices. Xiaomi/Aqara devices are for example known not to work with Zigbee router devices from Centralite, General Electrics, Iris, Ledvance/OSRAM, LIGHTIFY/Sylvania, Orvibo, PEQ, Securifi, and SmartThings/Samsung. Better results can usually be achieved by using mains-powered devices IKEA and Nue/3A Home or dedicated DIY routing devices based on Texas Instruments CC253x/CC26x2 and XBee Series 2/3 Zigbee radios. 
- Be patient as the pairing of some Zigbee devices may require multiple attempts and you may sometimes need to try again and again. 
    - Some devices, like example those from Xiaomi/Aqara, are known to not be 100% compliant with the standard Zigbee specifications and may therefore require many paring attempts over 10-20 minutes or longer. 

## External resources for initial Zigbee setup optimizations:

- [Zigbee and Wi-Fi Coexistence (Metageek - Advanced Wi-Fi Lessons and Training)](https://community.home-assistant.io/t/advice-on-zigbee-range-extending/175882/)
- [Choose your Zigbee channel wisely (Liam Alexander Colman - Home Assistant Guide)](https://home-assistant-guide.com/2020/10/29/choose-your-zigbee-channel-wisely/)
- [Smart Home Protocols: Zigbee Explained (Everything Smart Home on YouTube)](https://www.youtube.com/watch?v=UmpDXc3cXbU&ab_channel=EverythingSmartHome)
- [Improve Zigbee network performance and stability with these 3 tips (Home Automation Guy on YouTube)](https://www.youtube.com/watch?v=t-gw7kURXCk&t=2s&ab_channel=HomeAutomationGuy)
- [Zigbee 2022 TIPS with Home Assistant (digiblurDIY on YouTube)](https://www.youtube.com/watch?v=1dcAXkJxzcY&ab_channel=digiblurDIY)
- [How to improve your Zigbee network (Gadget-Freak)](https://gadget-freakz.com/how-to-improve-your-zigbee-network/)
- [Top 5 Reasons for Failures in Zigbee Deployments (NTS)](https://nts.com/services/education/zigbee-best-practices/)
- [Electrolama ZZH Adapter Troubleshooting and FAQ](https://electrolama.com/radio-docs/troubleshooting/)
- [2.4GHz Intra-System (or Self/Platform) Interference Demonstration (Unit 3 Compliance Blog)](https://www.unit3compliance.co.uk/2-4ghz-intra-system-or-self-platform-interference-demonstration/)
- [USB 3.0 Radio Frequency Interference Impact on 2.4 GHz Wireless Devices (Intel White Paper)](https://www.usb.org/sites/default/files/327216.pdf)
